### PR TITLE
Add mongodb to celery implementation.

### PR DESCRIPTION
**Pull request by l-vincent-l**
partial_fix #44 

With this we are able to retrieve the status and the result of a task.

### CODE REVIEW

File: lib/transport/application.ex [hunk 1]
```diff
@@ -31,4 +32,44 @@ defmodule Transport.Application do
     Endpoint.config_change(changed, removed)
     :ok
   end
+
+  defp get_mongodb_keywords!() do
```
Comment (maukoquiroga) [inline lib/transport/application.ex]:
```elixir
# config.exs
config :mongodb, mongodb_url: System.get_env("MONGODB_URL")

# here
Application.get_all_env(:mongodb)
```

File: lib/transport/application.ex [hunk 2]
```diff
@@ -31,4 +32,44 @@ defmodule Transport.Application do
...
+    "MONGODB_URL"
+    |> System.get_env()
+    |> case do
+      nil -> raise "Environment variable MONGODB_URL not found"
+      uri -> uri |> parse_mongodb_uri()
+    end
+  end
+
+  def parse_mongodb_uri(uri) do
+    uri_parsed = uri |> URI.parse()
```
Comment (maukoquiroga) [inline lib/transport/application.ex]: You don't need to pipe here

File: lib/transport/application.ex [hunk 3]
```diff
@@ -31,4 +32,44 @@ defmodule Transport.Application do
     Endpoint.config_change(changed, removed)
     :ok
   end
+
+  defp get_mongodb_keywords!() do
+    "MONGODB_URL"
+    |> System.get_env()
+    |> case do
+      nil -> raise "Environment variable MONGODB_URL not found"
+      uri -> uri |> parse_mongodb_uri()
```
Comment (maukoquiroga) [inline lib/transport/application.ex]: You don't need to pipe here

File: lib/transport/application.ex [hunk 4]
```diff
@@ -31,4 +32,44 @@ defmodule Transport.Application do
...
+      path when path in [nil, "", "/"] -> Map.put(uri_parsed, :database, nil)
+      path -> Map.put(uri_parsed, :database, String.trim_leading(path, "/"))
+    end
+  end
+
+  defp get_mongodb_username_password(uri_parsed) do
+    uri_parsed
+    |> Map.get(:userinfo)
+    |> case do
+      info when info in [nil, ""] -> Map.merge(uri_parsed,
```
Comment (maukoquiroga) [inline lib/transport/application.ex]: Better to add a new line, no?

File: lib/transport/data_validator/server.ex [hunk 5]
```diff
@@ -24,6 +24,10 @@ defmodule Transport.DataValidator.Server do
     GenServer.call(:publisher, {:apply_async, "tasks.perform", [args: [url]]})
   end
 
+  def get_task(task_id) do
+    GenServer.call(:publisher, {:get_task_meta, task_id})
```
Comment (maukoquiroga) [inline lib/transport/data_validator/server.ex]: Why `:get_task_meta`? That's celery's internal naming, for us is just a task.

File: lib/transport/data_validator/server.ex [hunk 6]
```diff
@@ -49,6 +53,22 @@ defmodule Transport.DataValidator.Server do
     end
   end
 
+  def handle_call({:get_task_meta, task_id}, _, state) do
+     :mongo
+     |> Mongo.find_one("celery_taskmeta", %{_id: task_id}, pool: DBConnection.Poolboy)
+     |> case do
+      nil -> {:reply, nil, state}
+      obj ->  {:reply, {:ok,
+                       %{:task_id   => obj["_id"],
```
Comment (maukoquiroga) [inline lib/transport/data_validator/server.ex]: I would create a struct ```Task``` for this, and define a ```find``` there.

File: test/transport/data_validator/server_test.exs [hunk 7]
```diff
@@ -1,8 +1,25 @@
 defmodule Transport.DataValidator.ServerTest do
   use ExUnit.Case, async: false
   alias Transport.DataValidator.Server
+  import Mock
 
   test "publish a message" do
     assert {:ok, _} = Server.validate_data("pouet")
   end
+
+  test "get taskmeta data" do
```
Comment (maukoquiroga) [inline test/transport/data_validator/server_test.exs]: I don't see the reason to mock this, as just doing a normal insert in the database before running the test will make it pass.

File: lib/transport/application.ex (hunk 1 above)
Comment (l-vincent-l) [reply lib/transport/application.ex]: done

File: lib/transport/data_validator/server.ex (hunk 5 above)
Comment (l-vincent-l) [reply lib/transport/data_validator/server.ex]: here we are doing celery internal things. I'm trying to stick to the original implementation to make it more easily debuggable.

File: lib/transport/celery_task.ex [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+         {:ok, %__MODULE__{
+             :task_id   => obj["_id"],
+             :status    => obj["status"],
+             :date_done => obj["date_done"],
+             :result    => result,
+             :traceback => traceback,
+             :children  => children
+         }}
+    else
+        error -> {:error, error}
```
Comment (maukoquiroga) [inline lib/transport/celery_task.ex]: Indentation shoulb de 2 spaces

File: lib/transport/celery_task.ex [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+  def apply(nil) do
+    {:error, "task not found"}
+  end
+
+  def apply(obj) do
+    with {:ok, result}    = Poison.decode(obj["result"]),
+         {:ok, traceback} = Poison.decode(obj["traceback"]),
+         {:ok, children}  = Poison.decode(obj["children"]) do
+         {:ok, %__MODULE__{
```
Comment (maukoquiroga) [inline lib/transport/celery_task.ex]: Indentation should be two spaces, relative to ```with```

**Pull request merged**